src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -2,44 +2,107 @@
 using System;
 using System.IO;
 using System.Management.Automation;
+using System.Management.Automation.Internal;
 
 namespace Microsoft.PowerShell.Commands
 {
     /// <summary>
-    /// The implementation of the "New-TemporaryFile" cmdlet
+    /// The implementation of the "New-TemporaryFile" cmdlet that has an optional 'Extension' Parameter property and a 'GuidBasedName' switch.
+    /// If this cmdlet errors then it throws a non-terminating error.
     /// </summary>
     [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=526726")]
     [OutputType(typeof(System.IO.FileInfo))]
     public class NewTemporaryFileCommand : Cmdlet
     {
         /// <summary>
-        /// returns a TemporaryFile
+        /// Specify a different file extension other than the default one, which is '.tmp'. The period in this parameter is optional.
+        /// </summary>
+        [Parameter(Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string Extension { get; set; } = ".tmp";
+
+        /// <summary>
+        /// Use a Guid as a file name to allow for more than 65,535 unique file names in the temporary folder.
+        /// </summary>
+        [Parameter()]
+        public SwitchParameter GuidBasedName  { get; set; }
+
+        /// <summary>
+        /// Returns a TemporaryFile.
         /// </summary>
         protected override void EndProcessing()
         {
+            // Check for invalid characters in extension
+            if (!string.IsNullOrEmpty(Extension) && Extension.IndexOfAny(Path.GetInvalidFileNameChars()) != -1)
```

Comment:
I guess it makes sense if we want to fix the problem somehow, but we just write a error message. So it's best to just follow CoreFX, where this check is done and just handle the exception. As a result, we will have only one block `try-catch` - see [my post](https://github.com/PowerShell/PowerShell/pull/4612#issuecomment-323577938)